core-ui/src/components/Extensibility/components/Columns.js

```diff
@@ -1,10 +1,20 @@
 import React from 'react';
 
 import { Widget } from './Widget';
+import { widgets } from '.';
+
+import './InlineDisplay.scss';
 
 export function Columns({ structure, ...props }) {
+  let classNames = 'panel-grid';
+  if (Array.isArray(structure.children)) {
+    if (structure.children.every(child => widgets[child?.widget]?.inline)) {
```

Comment:
I'm still not fond of this solution - as it doesn't check the context in which the component is used, it just checks what items are added in...
so if somebody added a block element (that still somehow renders correctly even inline), it will suddenly switch to full-fledged columns, instead of `inline-display`, and I don't think that's what we're aiming for.